src/main/java/reposense/wizard/OptionalPrompt.java

```diff
@@ -0,0 +1,35 @@
+package reposense.wizard;
+
+/**
+ * Represents an optional prompt that only calls the provided prompt if requested by the user.
+ */
+public class OptionalPrompt extends Prompt {
+    private final Prompt prompt;
+    private final static String YES_FLAG = "Y";
+    private final static String NO_FLAG = "N";
+    private final static String DESCRIPTION = String.format("%s/%s", YES_FLAG, NO_FLAG);
+
+    public OptionalPrompt(String question, Prompt prompt) {
+        super(question, DESCRIPTION);
+        this.prompt = prompt;
+    }
+
+    @Override
+    public InputBuilder addToInput(InputBuilder inputBuilder) {
+        return inputBuilder;
+    }
+
+    @Override
+    public Prompt[] run() {
+        if (getResponse().compareToIgnoreCase(YES_FLAG) == 0) {
+            return new Prompt[] {this.prompt};
+        }
+
+        if (getResponse().compareToIgnoreCase(NO_FLAG) == 0) {
+            return new Prompt[] {};
+        }
+
+        // Repeat OptionalPrompt until a valid flag is provided
+        return new Prompt[] {new OptionalPrompt(this.getQuestion(), this.prompt)};
```

Comment:
Instead of repeating Prompts, do you think it is better to just create a while loop until our condition is satisfied inside `run()`?

Reply:
Yeah sure, that is a good idea. I was excited to get a chance to use the `prompts.addFirst` method, but I think so I think it is indeed better to not keep creating new prompts objects. :P

Reply:
Update:
I realized why I did not do this initially, its because as prompt does not have access to the scanner, so it cannot call `promptAndGetInput` or `getInput`. Unless we change the design of prompt to include a scanner field, or for `run()` to take in a scanner as well, a while loop will not be possible. Let me know which way you prefer, I think that each of the ways has its own merits and demerits.

Reply:
Yea, in this case, I suppose passing `this` would be a better solution than having all Prompt take in a scanner.